### ISSUE TITLE
"sticky" filters

### ISSUE THREAD

**Issue by tkellen** [endpoints/endpoints#152]
Endpoints allows a server to specify both default includes and default filters for any endpoint.

Our current implementation of filters follows the JSON-API specification's semantics around `include`. That is, a server can specify a list of default included relations at a given endpoint, but if the user explicitly specifies any in their request, it should completely override the server-defaults. 

There are times when a server might wish to specify a filter that the user cannot opt out of (such as suppressing records which have been flagged as deleted). While one could argue this should be implemented at the database layer, the general question remains: servers should be able to specify filters that stay by default, and filters which cannot be altered. What should our API for this be?

The current API is this:
```js
controller.read({
  filter: {
    deleted: true
  }
});
```

Here are two proposed APIs:
```js
controller.read({
  filter: {
    deleted: {
      value: true,
      sticky: true,
      locked: true
    }
  }
});

controller.read({
  filter: {
    deleted: true
  },
  filterSettings: {
    deleted: {
      sticky: true,
      locked: true
    }
  }
});
```

I like the first best. Thoughts @bobholt?

**Comment by tkellen** [comment]
It's also just occurred to me that some of this might belong at the model layer (otherwise you'd have to repeatedly set your sticky/locked `deleted` filter at every controller which could include the relevant resources). Even so, an implementor probably needs the flexibility to control it on a per-endpoint basis.

**Comment by bobholt** [comment]
Yeah, from a per-endpoint standpoint, maybe I want to make an admin endpoint that can see the deleted records, so it's useful to do it in the model and use the controller for overrides.

I like the first one, too, for its simplicity. The object value is ugly, but I think it will be anyway. Here's another option to throw in the mix, though I still like number 1.

```
controller.read({
  filter: {
    employee: 12
  },
  stickyFilter: {
    is_active: true
  },
  lockedFilters: {
    deleted: true
  }
});
```

Granted, I'm not grokking the difference between sticky and locked. Would a filter need to be specified as both?

**Comment by tkellen** [comment]
locked means if the server default is, say `true` for some value, it's invalid to try to set it to anything else. The endpoint would either throw an error for that request or silently ignore the value.
